packages/select/custom-resources/AvOrganizationSelect.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  avFilteredOrganizationsApi,
+  avOrganizationsApi,
+} from '@availity/api-axios';
+import { ResourceSelect } from '..';
+
+const AvOrganizationSelect = ({
+  name,
+  parameters: { permissionId = [], resourceIds = [], regionId } = {},
+  ...props
+}) => {
+  // Convert permissions and resources to array so that the dependency array is kept consistent
+  permissionId =
+    typeof permissionId === 'string' ? [permissionId] : permissionId;
+  resourceIds = typeof resourceIds === 'string' ? [resourceIds] : resourceIds;
+
+  const OrganizationSelect = React.useMemo(
+    () =>
+      ResourceSelect.create({
+        resource:
+          resourceIds.length > 0 && permissionId.length > 0
+            ? avFilteredOrganizationsApi
+            : avOrganizationsApi,
+        labelKey: 'name',
+      }),
+    // React will not be able to statically check if values are the same
+    // but without spreading values, there would be new array references on each render,
+    // causing new api instances to be created each time
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [...resourceIds, ...permissionId]
+  );
+
+  return (
+    <OrganizationSelect
+      name={name}
+      parameters={{ permissionId, resourceIds, regionId }}
```

Comment:
`/organizations` does not have a `resourceIds` parameter. I think it would make more sense to make `resourceIds` its own prop and not include it in the `parameters`

Reply:
`/organizations` doesn't but `/axi-user-permissions` does, `avFilteredOrganizationsApi` will parse the query string from the `postGet` of `ResourceSelect` and break it up into separate calls to `/organizations` and `/axi-user-permissions`. Is there another way to get the `resourceIds` included in the request? I tried adding them as their own prop but I may have done it wrong

Reply:
The long term goal is hopefully to have some parameter added or a new resolver to some ms added that allows filtering orgs on resources. That being said if we make 1 unique parameter a prop here then its going to throw off the DX and logic some for both the open source code and the user implementing it.

It seems dirty to re-use the parameters and it does feel weird that its not a parameter of the organizations endpoint but rather the `API Resource` itself. Keeping it how we have it would not overcomplicate all the merging we do for the cache unique and parameters to be sent to the postGet for the time being until we have a better solution on the server side to handle resourceId filters.

Reply:
i think putting `resourceIds` in parameters throws off the DX. As a developer, the only way I know what i can send in my parameters is by checking what query parameters the api supports. But for `<AvOrganizationSelect />`, there would be this undocumented additional parameter that also isn't really a parameter because it doesn't get used in the api request like every other parameter. That's why i believe it should be separate, and documented.

Reply:
What would you think of modifying `ResourceSelect` to include something like `apiMethodParams` in the `postGet` request so that it is clearly separated? Or give `ResourceSelect` the ability to call api methods besides `postGet`, this way the method can be documented with the api and params can be passed together that way.

Reply:
`additionalPostGetArgs` on `<ResourceSelect />` seems like the way to go to me. `<AvOrganizationSelect />` accepts a `resourceIds` prop, and it renders a `<ResourceSelect />` with the `resourceIds` prop inside of its `additionalPostGetArgs`